### PR TITLE
Adding phrygian design ph100

### PR DESCRIPTION
**Pull request by phrygiandesign**
Adding the phrygian design ph100 into the mainline QMK repo

## Description

Addition of files and configuration for the [Phrygian Design ph100](https://phrygiandesign.com/products/ph100-mechanical-keyboard)

The ph100 board uses a STM32F042 microcontroller and provides 100 modular keyswitch positions.

I'm available to be contacted via github: @phrygiandesign or email: contact@phrygiandesign.com

### CODE REVIEW

File: keyboards/phrygian/ph100/readme.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+## Phrygian Design ph100
+
+![ph100](https://cdn.shopify.com/s/files/1/0470/8814/0443/products/DSCF0064_1080x.png)
+
+The ph100 is a 100-key modular keyboard from Phrygian Design.
+
+* Keyboard Maintainer: https://github.com/phrygiandesign
+* Hardware Supported: ph100 PCB
+* Hardware Availability: https://phrygiandesign.com/products/ph100-mechanical-keyboard
+Keyboard support: contact@phrygiandesign.com
```
Comment (drashna) [inline keyboards/phrygian/ph100/readme.md:10]:
```suggestion
* Keyboard support: contact@phrygiandesign.com
```

File: keyboards/phrygian/ph100/rules.mk [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# MCU name
+MCU = STM32F042
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = full     # Virtual DIP switch configuration
```
Comment (drashna) [inline keyboards/phrygian/ph100/rules.mk:7]:
```suggestion
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by phrygiandesign** [comment]
Commit your suggested changes, thanks for getting back quick btw.

File: keyboards/phrygian/ph100/rules.mk [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# MCU name
+MCU = STM32F042
+
+# Build Options
```
Comment (drashna) [inline keyboards/phrygian/ph100/rules.mk:4]:
```suggestion
# Bootloader selection
BOOTLOADER = stm32-dfu

# Build Options
```

**Review by drashna — APPROVED** [review]

File: keyboards/phrygian/ph100/rules.mk [hunk 4]
```diff
@@ -0,0 +1,26 @@
+# MCU name
+MCU = STM32F042
+
+# Bootloader selection
+BOOTLOADER = stm32-dfu
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
```
Comment (drashna) [inline keyboards/phrygian/ph100/rules.mk:10]:
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

File: keyboards/phrygian/ph100/rules.mk [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (drashna) [inline keyboards/phrygian/ph100/rules.mk:21]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/phrygian/ph100/bootloader_defs.h [hunk 6]
```diff
@@ -0,0 +1,7 @@
+/* Address for jumping to bootloader on STM32 chips. */
```
Comment (fauxpark) [inline keyboards/phrygian/ph100/bootloader_defs.h:1]: This file can be deleted.

File: keyboards/phrygian/ph100/info.json [hunk 7]
```diff
@@ -0,0 +1,113 @@
+{
+    "keyboard_name": "ph100",
+    "url": "https://phrygiandesign.com/products/ph100-mechanical-keyboard",
+    "maintainer": "phrygiandesign",
+    "width": 21.5,
+    "height": 6,
```
Comment (fauxpark) [inline keyboards/phrygian/ph100/info.json:6]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/phrygian/ph100/bootloader_defs.h (hunk 6 above)
Comment (phrygiandesign) [reply keyboards/phrygian/ph100/bootloader_defs.h]: Deleted and committed to master

**Comment by phrygiandesign** [comment]
Committed all suggestions, let me know if you need anything else. Keen to get this pulled finally

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**